### PR TITLE
UI for form submission

### PR DESCRIPTION
**Pull request by lujasmine**
- Added the submit form route that renders the `SubmitForm` component 
- Created a generic `FormField` component that renders a `select` or `input` type field and passes on the prop types
- Validation functions for each field
- Styling tidy up

Next: Need to add the react-query mutation that will call the post route to submit the new user

Note: Screenshot added in ticket

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+import {
+  Route,
+  Routes,
+  Navigate,
+  BrowserRouter as Router,
+} from "react-router-dom";
 
-import Home from "./pages/Home";
+import SubmitForm from "./pages/SubmitForm";
+import UsersList from "./pages/UsersList";
```
Comment (8rianHo) [inline src/App.jsx:12]:
```suggestion
import UsersList from "./pages/UsersList";
import SubmitForm from "./pages/SubmitForm";
```

to follow your standard import rules

File: src/App.jsx [hunk 2]
```diff
@@ -18,11 +25,17 @@ const App = () => {
...
-            {/* TODO tbc route for form submission */}
-            <Route path="/submit"></Route>
-          </Routes>
+          <VStack w={"80%"} mx={"auto"}>
+            <Box w={"100%"} textAlign={"left"} my={4}>
+              <Heading textAlign={"left"}>Vietnamese Directory</Heading>
+            </Box>
+            <Routes>
+              <Route path="/" element={<UsersList />} />
+              <Route path="/submit" element={<SubmitForm />}></Route>
```
Comment (8rianHo) [inline src/App.jsx:34]:
```suggestion
              <Route path="/submit" element={<SubmitForm />} />
```

should work?

File: src/components/FormField.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  >
+    {fieldProps.options.map((ageOption) => (
+      <option value={ageOption.value} key={ageOption.value}>
+        {ageOption.label}
+      </option>
+    ))}
+  </Select>
+);
+
+const _showField = (fieldType, field, fieldProps) => {
```
Comment (8rianHo) [inline src/components/FormField.jsx:34]: haven't seen this before: what does the _ mean?

File: src/components/UserCard.jsx [hunk 4]
```diff
@@ -1,46 +1,41 @@
...
-            textTransform="uppercase"
-          >
-            Location: {user.location.city}
-          </Box>
+        <Text
+          color="gray.500"
+          letterSpacing="wide"
+          fontSize={"s"}
+          fontWeight={"semibold"}
+          textTransform="uppercase"
```
Comment (8rianHo) [inline src/components/UserCard.jsx:31]: can we add the square brackets to cooler, letterSpacing and textTransform please?

File: src/components/UserCard.jsx [hunk 5]
```diff
@@ -1,46 +1,41 @@
...
 
-          <Text mt={1} fontWeight={"medium"}>
-            Software Engineer @ Google
-          </Text>
-        </Box>
+        <Text fontWeight={"medium"}>Software Engineer @ Google</Text>
 
-        <HStack mt={4} spacing={2} align={"center"}>
+        <HStack spacing={2} align={"center"}>
           <Button colorScheme="black" variant="outline">
```
Comment (8rianHo) [inline src/components/UserCard.jsx:39]: missed this but looks like there are some inconsistencies with the prop values 😃 not a big deal but can we add the square brackets

tbf there should be a linting rule for chakra

File: src/pages/SubmitForm.jsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+          console.log(values);
+          setTimeout(() => {
+            alert(JSON.stringify(values, null, 2));
+            actions.setSubmitting(false);
+          }, 1000);
+        }}
+      >
+        {(props) => (
+          <Form>
+            <VStack spacing={4} alignItems="left">
```
Comment (8rianHo) [inline src/pages/SubmitForm.jsx:36]:
```suggestion
            <VStack spacing={4} alignItems={"left"}>
```

File: src/pages/UsersList.jsx [hunk 7]
```diff
@@ -19,7 +17,7 @@ const UsersList = () => {
   }
 
   return (
-    <SimpleGrid minChildWidth="320px" spacing="40px">
+    <SimpleGrid w={"100%"} minChildWidth="320px" spacing="40px" py={5}>
```
Comment (8rianHo) [inline src/pages/UsersList.jsx:20]:
```suggestion
    <SimpleGrid w={"100%"} minChildWidth={"320px"} spacing={"40px"} py={5}>
```

**Review by 8rianHo — APPROVED** [review]
not really blockerz

File: src/components/FormField.jsx (hunk 3 above)
Comment (lujasmine) [reply src/components/FormField.jsx:34]: It doesn't add any functionality in javascript, but a convention in some other languages used to denote private functions that aren't imported!

**Pull request merged**